app/models/portfolio.rb

```diff
@@ -24,7 +25,20 @@ def add_portfolio_item(portfolio_item)
   end
 
   def metadata
-    {:user_capabilities => user_capabilities,
-     :shared            => self.access_control_entries.any?}
+    ancillary_metadata.metadata_attributes.merge(:user_capabilities => user_capabilities)
+  end
+
+  private
+
+  def update_ancillary_metadata
+    ancillary_metadata.statistics = statistics_metadata
+  end
+
+  def statistics_metadata
+    {
+      'approval_processes' => tags.where(:name => 'workflows', :namespace => 'approval').count,
+      'portfolio_items'    => portfolio_items.count,
+      'shared_groups'      => access_control_entries.select(:group_uuid).distinct.count
```

Comment:
@gmcculloug with some of the redesign that happened with the ACE table now there is one group for each portfolio and the permissions moved into a separate table. In the earlier design we used to have multiple rows with (group_uuid, portfolio_id, permission) and we needed the distinct. We can remove the distinct here